src/main/php/CodacyPDepend/index.php

```diff
@@ -0,0 +1,180 @@
+        pushToMapValueArray($map, $key, $function);
+    }
+
+    return $map;
+}
+
+function filenameToMethods($node)
+{
+    $map = new Map();
+    foreach ($node->getClasses() as $content) {
+        $key = getFilename($content);
+        concatMapValueArray($map, $key, $content->getMethods());
+    }
+    return $map;
+}
+
+function contentMapToFileComplexities($content_map, $cyclomaticAnalyzer)
+{
+    $res = new Map();
+    foreach ($content_map as $file => $nodes) {
+        $res[$file] = genericMap(
+            function ($node) use ($cyclomaticAnalyzer) {
+                $ccn = $cyclomaticAnalyzer->getCcn($node);
+                $line = $node->getStartLine();
+                return new LineComplexity($line, $ccn);
+            },
+            $nodes
+        );
+    }
+    return $res;
+}
+
+function filesToNrClasses($result)
+{
+    $res = new Map();
+    foreach ($result as $node) {
+        foreach ($node->getClasses() as $class) {
+            $file = getFilename($class);
+            if ($res->hasKey($file)) $res[$file]++;
+            else $res[$file] = 1;
+        }
+    }
+    return $res;
+}
+
+function filesToNrMethods($result)
+{
+    $res = new Map();
+    foreach ($result as $node) {
+        foreach ($node->getClasses() as $class) {
+            $file = getFilename($class);
+            if ($res->hasKey($file)) $res[$file] += $class->getMethods()->count();
+            else $res[$file] = $class->getMethods()->count();
+        }
+        foreach ($node->getFunctions() as $function) {
+            $file = getFilename($function);
+            if ($res->hasKey($file)) $res[$file] += 1;
+            else $res[$file] = 1;
+        }
+    }
+    return $res;
+}
+
+function resultToContentMap($result)
+{
+    $content_map = new Map();
+    foreach ($result as $node) {
+        $files_to_methods = filenameToMethods($node);
+        $files_to_functions = filenameToFunctions($node);
+        foreach (array($files_to_methods, $files_to_functions) as $map) {
+            mergeMapsOfArrays($content_map, $map);
+        }
+    }
+    return $content_map;
+}
+
+function stripStringPrefix($str, $prefix)
+{
+    return substr($str, 0, strlen($prefix)) == $prefix ? substr($str, strlen($prefix)) : $str;
+}
+
+try {
+    error_reporting(E_ERROR);
+    $app = new Application();
+    $engine = $app->getEngine();
+    $codacyReportGenerator = new CodacyReportGenerator();
+    $engine->addReportGenerator($codacyReportGenerator);
+    addFilesFromConfiguration($engine);
+    $result = $engine->analyze();
+    $cyclomaticAnalyzer = $codacyReportGenerator->getCyclomaticComplexityAnalyzer();
+    $content_map = resultToContentMap($result);
+    $fileToLineComplexities = contentMapToFileComplexities($content_map, $cyclomaticAnalyzer);
+
+    $filesToNrClasses = filesToNrClasses($result);
+    $filesToNrMethods = filesToNrMethods($result);
+
+    $files = $filesToNrClasses->keys()->union($filesToNrMethods->keys());
+
+    foreach ($files as $file) {
```

Comment:
we are already iterating through the files, isn't it?
`addDirectoryRecursively`

Reply:
You mean reuse the code from `addDirectoryRecursively`? Here we are iterating only on files that have complexity results, not on all of them

Reply:
fair enough